modules/openapi-generator/src/main/resources/typescript/configuration.mustache

```diff
@@ -105,43 +105,36 @@ export function createConfiguration(conf: ConfigurationParameters = {}): Configu
  */
 {{#useInversify}}
 export function mergeConfiguration(conf: Configuration, options?: Configuration): Configuration {
-    if (options) {
-        conf = options;
-    }
-    return conf;
+    return options || conf;
 }
 {{/useInversify}}
 {{^useInversify}}
 export function mergeConfiguration(conf: Configuration, options?: ConfigurationOptions): Configuration {
-    let allMiddleware: Middleware[] = [];
-    if (options && options.middleware) {
-        const middlewareMergeStrategy = options.middlewareMergeStrategy || "replace" // default to replace behavior
-        // call-time middleware provided
-        const calltimeMiddleware: Middleware[] = options.middleware;
+    if (!options) {
+        return conf;
+    }
+    return {
+        baseServer: options.baseServer || conf.baseServer,
+        httpApi: options.httpApi || conf.httpApi,
+        authMethods: options.authMethods || conf.authMethods,
+        middleware: mergeMiddleware(conf.middleware, options?.middleware, options?.middlewareMergeStrategy),
+    };
+}
 
-        switch(middlewareMergeStrategy) {
+function mergeMiddleware(staticMiddleware: Middleware[], calltimeMiddleware?: Middleware[], strategy?: "append" | "prepend" | "replace") {
+    if (!calltimeMiddleware) {
+        return staticMiddleware;
+    }
+    // default to replace behavior
+    switch(strategy ?? "replace") {
         case "append":
-            allMiddleware = conf.middleware.concat(calltimeMiddleware);
-            break;
+            return staticMiddleware.concat(calltimeMiddleware);
         case "prepend":
-            allMiddleware = calltimeMiddleware.concat(conf.middleware)
-            break;
+            return calltimeMiddleware.concat(staticMiddleware)
         case "replace":
-            allMiddleware = calltimeMiddleware
-            break;
-        default:
-            throw new Error(`unrecognized middleware merge strategy '${middlewareMergeStrategy}'`);
-        }
-    }
-    if (options) {
-        conf = {
-          baseServer: options.baseServer || conf.baseServer,
-          httpApi: options.httpApi || conf.httpApi,
-          authMethods: options.authMethods || conf.authMethods,
-          middleware: allMiddleware || conf.middleware
-        };
+            return calltimeMiddleware
     }
-    return conf;
+    throw new Error(`Unrecognized middleware merge strategy '${strategy}'`)
```

Comment:
did you move this out because of the explicit return rule? Otherwise I think it was semantically easier to understand inside the switch.

Reply:
It felt cleaner to me, because it made it obvious, that it is a last-resort for the function (actually in typescript it is unreachable anyway, because the switch completely covers all possible values of strategy)
Putting it into `default` makes sense as well because it is concerned with invalid values for `strategy`.
I can put it back into `default:` if you prefer.

Reply:
Makes sense. I moved it back inside.